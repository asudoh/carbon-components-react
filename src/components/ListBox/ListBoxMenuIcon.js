import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';
import { iconCaretDown } from 'carbon-icons';
import Icon from '../Icon';

export const translationIds = {
  'close.menu': 'close.menu',
  'open.menu': 'open.menu',
};

const defaultTranslations = {
  [translationIds['close.menu']]: 'Close menu',
  [translationIds['open.menu']]: 'Open menu',
};

/**
 * `ListBoxMenuIcon` is used to orient the icon up or down depending on the
 * state of the menu for a given `ListBox`
 */
const ListBoxMenuIcon = ({ isOpen, prefix, translateWithId: t }) => {
  const className = cx({
    [`${prefix}--list-box__menu-icon`]: true,
    [`${prefix}--list-box__menu-icon--open`]: isOpen,
  });
  const description = isOpen ? t('close.menu') : t('open.menu');
  return (
    <div className={className}>
      <Icon icon={iconCaretDown} description={description} alt={description} />
    </div>
  );
};

ListBoxMenuIcon.propTypes = {
  /**
   * Specify whether the menu is currently open, which will influence the
   * direction of the menu icon
   */
  isOpen: PropTypes.bool.isRequired,

  /**
   * The selector prefix
   */
  prefix: PropTypes.string,

  /**
   * i18n hook used to provide the appropriate description for the given menu
   * icon. This function takes in an id defined in `translationIds` and should
   * return a string message for that given message id.
   */
  translateWithId: PropTypes.func.isRequired,
};

ListBoxMenuIcon.defaultProps = {
  prefix: 'bx',
  translateWithId: id => defaultTranslations[id],
};

export default ListBoxMenuIcon;
