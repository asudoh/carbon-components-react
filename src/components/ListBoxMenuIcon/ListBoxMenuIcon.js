import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Icon from '../Icon';

const ListBoxMenuIcon = ({ isOpen }) => {
  const className = cx({
    'bx--list-box__menu-icon': true,
    'bx--list-box__menu-icon--open': isOpen,
  });
  const description = isOpen ? 'Close menu' : 'Open menu';
  return (
    <Icon className={className} name="caret--down" description={description} />
  );
};

ListBoxMenuIcon.propTypes = {
  isOpen: PropTypes.bool.isRequired,
};

export default ListBoxMenuIcon;
