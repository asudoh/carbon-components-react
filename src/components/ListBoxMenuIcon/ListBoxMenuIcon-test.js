import React from 'react';
import { shallow } from 'enzyme';
import ListBoxMenuIcon from '../ListBoxMenuIcon';

describe('ListBoxMenuIcon component', () => {
  it('should render', () => {
    const openWrapper = shallow(<ListBoxMenuIcon isOpen={true} />);
    const closeWrapper = shallow(<ListBoxMenuIcon isOpen={false} />);
    expect(openWrapper).toMatchSnapshot();
    expect(closeWrapper).toMatchSnapshot();
  });

  it('should update the Icon description if the menu is open or closed', () => {
    const wrapper = shallow(<ListBoxMenuIcon isOpen={false} />);
    expect(wrapper.prop('description')).toBe('Open menu');
    wrapper.setProps({ isOpen: true });
    expect(wrapper.prop('description')).toBe('Close menu');
  });
});
