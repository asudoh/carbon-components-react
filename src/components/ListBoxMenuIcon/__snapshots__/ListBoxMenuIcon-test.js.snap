// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBoxMenuIcon component should render 1`] = `
<Icon
  className="bx--list-box__menu-icon bx--list-box__menu-icon--open"
  description="Close menu"
  fillRule="evenodd"
  name="caret--down"
  role="img"
/>
`;

exports[`ListBoxMenuIcon component should render 2`] = `
<Icon
  className="bx--list-box__menu-icon"
  description="Open menu"
  fillRule="evenodd"
  name="caret--down"
  role="img"
/>
`;
